src/web/mjs/connectors/Mangafest.mjs

```diff
@@ -0,0 +1,94 @@
+import Connector from '../engine/Connector.mjs';
+import Manga from '../engine/Manga.mjs';
+
+export default class Mangafest extends Connector {
+
+    constructor() {
+        super();
+        super.id = 'mangafest';
+        super.label = 'Mangafest X';
+        this.tags = [ 'indonesian', 'webtoon'];
+        this.url = 'https://mangafest.net';
+    }
+
+    async _getMangas() {
+        let request = new Request(new URL('/category/manga/', this.url), this.requestOptions);
+        let data = await this.fetchDOM(request, 'nav.navigation.pagination div.nav-links a.page-numbers:not(.next)');
+        const pages = parseInt(data[data.length-1].pathname.match(/\d+\/$/));
+
+        let mangas = [];
+        for (let page = 1; page <= pages; page++) {
+            request = new Request(new URL('/category/manga/page/'+page+'/', this.url), this.requestOptions);
+            data = await this.fetchDOM(request, 'div.post_image a');
+            mangas.push(...data.map(manga => {
+                return {
+                    id: this.getRootRelativeOrAbsoluteLink(manga.pathname, request.url),
+                    title: manga.getAttribute('aria-label').trim()
+                };
+            }));
+        }
+
+        return mangas;
+    }
+
+    async _getChapters(manga) {
+        // website provides mangas as whole
+        return [
+            {
+                id: manga.id,
+                title: manga.title
+            }
+        ];
+    }
+
+    async _getPages(manga) {
+        let request = new Request(new URL(manga.id, this.url), this.requestOptions);
+        let data = await this.fetchDOM(request, 'div#ngy2p');
+        const flickr_data = JSON.parse(data[0].getAttribute('data-nanogallery2-portable'));
+
+        let url = new URL('https://api.flickr.com/services/rest/?&method=flickr.photosets.getPhotos&api_key=2f0e634b471fdb47446abcb9c5afebdc&format=json');
+        url.searchParams.set('user_id', flickr_data.userID);
+        url.searchParams.set('photoset_id', flickr_data.photoset);
+        url.searchParams.set('page', 1); // Returns 500 pictures per page. Since these are hobbyist mangas I assume none will be over 500 pages.
+
+        request = new Request(url);
+        data = await this.fetchRegex(request, /^jsonFlickrApi\((.*)\)/g);
```

Comment:
Is greedy intentional?
It may swallow any closing brackets into the group.

When dealing non-greedy (aka lazy) cases i prefer to define the ending character by _all characters but this_ regex (as an alternative to the lazy operator)
e.g. `/^jsonFlickrApi\(([^)]+)\)/g`

https://regex101.com/r/vFON21/2
https://regex101.com/r/vFON21/1

Reply:
Yes but i added a `$` as last character in the meantime. Flickr always wraps the response in `jsonFlickrApi(...)` for whatever reason. So i now use `/^jsonFlickrApi\((.*)\)$/` which will prevent consuming the closing bracket.